### PR TITLE
SWDEV-326661 - device_lib_path, return error code when build fails

### PR DESCRIPTION
**Pull request by agunashe**
hipcc cpp to return failure code.  Fix to use devicelibpath when env variable is available 

Change-Id: Idca31dedf3b5ce1524fc344a47a3c96afde562ad

### CODE REVIEW

File: src/hipBin_amd.h [hunk 1]
```diff
@@ -359,12 +359,12 @@ string HipBinAmd::getCppConfig() {
...
 
 string HipBinAmd::getDeviceLibPath() const {
-  string deviceLibPath;
   const EnvVariables& var = getEnvVariables();
   const string& rocclrHomePath = getRocclrHomePath();
   const string& roccmPath = getRoccmPath();
   fs::path bitCodePath = rocclrHomePath;
   bitCodePath /= "lib/bitcode";
+  string deviceLibPath = var.deviceLibPathEnv_;
   if (var.deviceLibPathEnv_.empty() && fs::exists(bitCodePath)) {
```
Comment (searlmc1) [inline src/hipBin_amd.h:368]: nit: deviceLibPath.empty() rather than var.deviceLibPathEnv_.empty() ?

**Review by searlmc1 — COMMENTED** [review]
See comment on src/hipBin_amd.h line 368

File: src/hipBin_amd.h (hunk 1 above)
Comment (agunashe) [reply src/hipBin_amd.h]: Fixed in the latest commit - https://github.com/ROCm-Developer-Tools/HIPCC/pull/19/commits/f489dff2c6ffc3c90adcd853b091d178079d96ab

**Review by searlmc1 — APPROVED** [review]
LGTM

**Comment by agunashe** [comment]
Request to please merge this PR to develop branch. I do not have write access to merge this change

**Pull request merged**